services/web-app/components/color-token-table/color-token-table.js

```diff
@@ -0,0 +1,404 @@
+        this.hexToRgb(hex).b
+      }, 0.5)`
+    }
+    return (
+      <div className={styles['color-token-value']}>
+        <ul>
+          <li>{value[currentTheme].name}</li>
+          <li>—</li>
+          <li>{value[currentTheme].hex}</li>
+        </ul>
+        <div>
+          <div
+            className={styles['color-token-value__block']}
+            style={{
+              backgroundColor: bgColor,
+              border: value[currentTheme].hex === '#ffffff' && '1px solid #BEBEBE'
+            }}
+          />
+          <OverflowMenu floatingMenu={false} flipped>
+            <CopyToClipboard text={value[currentTheme].hex}>
+              <OverflowMenuItem primaryFocus itemText="Copy hex" />
+            </CopyToClipboard>
+            <CopyToClipboard text={token}>
+              <OverflowMenuItem itemText="Copy token" />
+            </CopyToClipboard>
+          </OverflowMenu>
+        </div>
+      </div>
+    )
+  }
+
+  renderToken = (token, tokenInfo, key) => {
+    const roles = tokenInfo.role.map((role, i) => (
+      <li key={i}>
+        {role}
+        {i !== tokenInfo.role.length - 1 && ';'}
+      </li>
+    ))
+    return (
+      <tr key={key}>
+        <td>
+          <code>{token}</code>
+        </td>
+        <td>
+          <ul>{roles}</ul>
+        </td>
+        <td>{this.renderValue(token, tokenInfo)}</td>
+      </tr>
+    )
+  }
+
+  render() {
+    const themeSwitcherClasses = clsx(styles['theme-switcher'], {
+      [styles['theme-switcher--stuck']]: this.state.sticky
+    })
+
+    return (
+      <Grid condensed className={clsx(styles['color-token-table'])}>
+        <Column sm={4} md={8} lg={16}>
+          <StickyContainer navBar banner secondary={false}>
+            <ContentSwitcher className={themeSwitcherClasses} onChange={this.switchTheme}>
+              <Switch name="white" text={this.state.mobile ? 'Wte' : 'White'} />
+              <Switch name="g10" text={this.state.mobile ? 'G10' : 'Gray 10'} />
+              <Switch name="g90" text={this.state.mobile ? 'G90' : 'Gray 90'} />
+              <Switch name="g100" text={this.state.mobile ? 'G100' : 'Gray 100'} />
+            </ContentSwitcher>
+          </StickyContainer>
+          <section>
+            <H3>Background</H3>
+
+            <Grid condensed>
+              <Column sm={4} md={8} lg={16}>
+                <table className={tableStyles['page-table']}>
+                  <thead>
+                    <tr>
+                      <th>Token</th>
+                      <th>Role</th>
+                      <th>Value</th>
+                    </tr>
+                  </thead>
+                  <tbody>
+                    {Object.keys(colorTokens['background-tokens']).map((token, i) =>
+                      this.renderToken(token, colorTokens['background-tokens'][token], i)
+                    )}
+                  </tbody>
+                </table>
+              </Column>
+            </Grid>
```

Comment:
It looks like this block could be extracted into its own component that takes a few props to avoid duplication across this file.

Reply:
Agreed, saw that sonarcloud flagged this 👍